homeassistant/components/evohome/__init__.py

```diff
@@ -422,20 +423,20 @@ async def save_auth_tokens(self) -> None:
 
         await self._store.async_save(app_storage)
 
-    async def call_client_api(self, api_function, refresh=True) -> Any:
-        """Call a client API."""
+    async def call_client_api(self, api_function, update_state=True) -> Any:
+        """Call a client API and update the broker state if required."""
         try:
             result = await api_function
         except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
             if not _handle_exception(err):
                 return
 
-        if refresh:
-            self.hass.helpers.event.async_call_later(1, self.async_update())
+        if update_state:
+            asyncio.create_task(self.async_update(manual_update=True))
```

Comment:
This would have been enough to fix the bug 😉 

```python
            self.hass.helpers.event.async_call_later(1, self.async_update)
```

But this will work too.

Reply:
OK, best of both worlds:
```
self.hass.helpers.event.async_call_later(1, self._update_v2_api_state)
```